pkg/controller/datavolume-controller.go

```diff
@@ -1122,73 +1171,56 @@ func (r *DatavolumeReconciler) getSnapshotClassForSmartClone(dataVolume *cdiv1.D
 		if k8serrors.IsNotFound(err) {
-			r.log.V(3).Info("Source PVC is missing", "source namespace", sourcePvcSpec.Namespace, "source name", sourcePvcSpec.Name)
+			log.Info("Source PVC is missing", "source namespace", sourcePvcSpec.Namespace, "source name", sourcePvcSpec.Name)
+			return false, nil
 		}
-		return "", errors.New("source PVC not found")
+		return false, err
 	}
 
 	targetPvcStorageClassName := targetStorageSpec.StorageClassName
 	targetStorageClass, err := GetStorageClassByName(r.client, targetPvcStorageClassName)
 	if err != nil {
-		return "", err
+		return false, err
 	}
 	if targetStorageClass == nil {
-		r.log.V(3).Info("Target PVC's Storage Class not found")
-		return "", errors.New("Target PVC storage class not found")
+		log.Info("Target PVC's Storage Class not found")
+		return false, nil
 	}
 	targetPvcStorageClassName = &targetStorageClass.Name
 	sourcePvcStorageClassName := pvc.Spec.StorageClassName
 
 	// Compare source and target storage classess
 	if *sourcePvcStorageClassName != *targetPvcStorageClassName {
-		r.log.V(3).Info("Source PVC and target PVC belong to different storage classes", "source storage class",
+		log.Info("Source PVC and target PVC belong to different storage classes", "source storage class",
 			*sourcePvcStorageClassName, "target storage class", *targetPvcStorageClassName)
-		return "", errors.New("source PVC and target PVC belong to different storage classes")
+		return false, nil
 	}
 
 	sourceVolumeMode := resolveVolumeMode(pvc.Spec.VolumeMode)
 	targetSpecVolumeMode, err := getStorageVolumeMode(r.client, dataVolume, targetStorageClass)
 	if err != nil {
-		return "", err
+		return false, err
 	}
 	targetVolumeMode := resolveVolumeMode(targetSpecVolumeMode)
 	if sourceVolumeMode != targetVolumeMode {
-		r.log.V(3).Info("Source PVC and target PVC have different volume modes, falling back to host assisted clone", "source volume mode",
+		log.Info("Source PVC and target PVC have different volume modes, falling back to host assisted clone", "source volume mode",
 			sourceVolumeMode, "target volume mode", targetVolumeMode)
-		return "", errors.New("Source PVC and target PVC have different volume modes, falling back to host assisted clone")
+		return false, nil
 	}
 
-	// Fetch the source storage class
 	srcStorageClass := &storagev1.StorageClass{}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: *sourcePvcStorageClassName}, srcStorageClass); err != nil {
-		r.log.V(3).Info("Unable to retrieve storage class, falling back to host assisted clone", "storage class", *sourcePvcStorageClassName)
-		return "", errors.New("unable to retrieve storage class, falling back to host assisted clone")
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: *targetPvcStorageClassName}, srcStorageClass); err != nil {
+		log.Info("Unable to retrieve storage class, falling back to host assisted clone", "storage class", *targetPvcStorageClassName)
+		return false, nil
```

Comment:
The same question here, shouldn't it return en err, so the operation can be retried again?

Reply:
Good point, I think I agree.